src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java

```diff
@@ -85,23 +85,22 @@ public static RequestRepository using(org.folio.circulation.support.Clients clie
 
   public CompletableFuture<Result<MultipleRecords<Request>>> findBy(String query) {
     return requestsStorageClient.getManyWithRawQueryStringParameters(query)
-      .thenApply(flatMapResult(this::mapResponseToRequests))
-      .thenComposeAsync(result -> itemRepository.fetchItemsFor(result, Request::withItem))
+      .thenApply(flatMapResult(this::mapResponseToRequests)).thenComposeAsync(this::fetchFields);
+  }
```

Comment:
try something like this: 

 return requestsStorageClient.getManyWithRawQueryStringParameters(query)
      .thenApply(flatMapResult(this::mapResponseToRequests))
      .thenComposeAsync(r -> r.after(this::fetchFields));

Reply:
We can't do such, since in` after(new Function<MultipleRecords<Request>, CompletableFuture<Result<MultipleRecords<Request>>>>()) ` and in the fetchFields() method has fetchItemFor by itemRepository, it needs to get `Result<MultipleRecords<Request>>` as input, but not just `MultipleRecords<Request>` 

To be more understandable, please look at the original method on the tlr-feature branch

Reply:
It works for me:

![image](https://user-images.githubusercontent.com/53909129/148741993-ec66429f-5c73-48d0-854b-6ca1ff377a7e.png)

Reply:
fixed